plugin-gradle/src/main/java/com/diffplug/gradle/spotless/JavaExtension.java

```diff
@@ -83,16 +83,23 @@ public GoogleJavaFormatConfig googleJavaFormat() {
 	 * for an workaround for using snapshot versions.
 	 */
 	public GoogleJavaFormatConfig googleJavaFormat(String version) {
+		return googleJavaFormat(GoogleJavaFormatStep.defaultArtifact(), version);
+	}
+
+	public GoogleJavaFormatConfig googleJavaFormat(String artifact, String version) {
+		Objects.requireNonNull(artifact);
 		Objects.requireNonNull(version);
-		return new GoogleJavaFormatConfig(version);
+		return new GoogleJavaFormatConfig(artifact, version);
 	}
 
 	public class GoogleJavaFormatConfig {
+		final String artifact;
```

Comment:
This should be `String mavenCoordinate = GoogleJavaFormatStep.MAVEN_COORDINATE`, and there should be a method `GoogleJavaFormatConfig mavenCoordinate(String groupArtifact)`

Reply:
Same rationale as the previous. Is the name `mavenCoordinate` favored over `groupArtifact`?

Reply:
If you don't like the (user sets either `version` or `mavenCoordinate`) suggestion above, I'm also okay with `groupIdArtifactId` or `groupArtifact`.

Reply:
As mentioned above, probably `customMavenCoordinate` or `customGroupArtifact` would be better.

Reply:
I think `groupArtifact` without `custom` is good enough. It IS custom when user invoke the corresponding method.